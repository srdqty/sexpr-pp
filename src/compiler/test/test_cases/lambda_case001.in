(lambda () 5)
(lambda a a)
(lambda (a) a)
(lambda (a b) a b)
(lambda (a . b) a b)
(lambda (a b . c) a b c)
(lambda (a b c) (lambda (d e f) (lambda (g h i) a b c d e f g h i)))
