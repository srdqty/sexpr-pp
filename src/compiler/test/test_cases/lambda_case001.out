(FunFixArity (Arity = 0) (Constant 5))
(FunVarArity (Arity >= 0) (ShallowRef 0))
(FunFixArity (Arity = 1) (ShallowRef 0))
(FunFixArity (Arity = 2) (ShallowRef 0) (ShallowRef 1))
(FunVarArity (Arity >= 1) (ShallowRef 0) (ShallowRef 1))
(FunVarArity (Arity >= 2) (ShallowRef 0) (ShallowRef 1) (ShallowRef 2))
(FunFixArity (Arity = 3) (FunFixArity (Arity = 3) (FunFixArity (Arity = 3) (DeepRef 2 0) (DeepRef 2 1) (DeepRef 2 2) (DeepRef 1 0) (DeepRef 1 1) (DeepRef 1 2) (ShallowRef 0) (ShallowRef 1) (ShallowRef 2))))
